testsuite/tests/singlecluster/gateway/reconciliation/change_targetref/test_update_dnspolicy_target_ref.py

```diff
@@ -0,0 +1,35 @@
+"""
+Test for changing targetRef field in DNSPolicy
+"""
+
+import pytest
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.dnspolicy]
+
+
+@pytest.fixture(scope="module")
+def authorization():
+    """
+    Override the authorization fixture to prevent the creation of an AuthPolicy.
+    This ensures no authentication is enforced during the test
+    """
+    return None
+
+
+def test_update_dns_policy_target_ref(
+    route2, tls_policy, tls_policy2, gateway, gateway2, dns_policy, client, client2, change_target_ref
+):  # pylint: disable=unused-argument
+    """Test updating the targetRef of an DNSPolicy from Gateway 1 to Gateway 2"""
+    assert gateway.refresh().is_affected_by(dns_policy)
```

Comment:
```suggestion
    assert gateway.wait_until(lambda obj: not obj.is_affected_by(dns_policy))
```
is more stable, it won't fail immediately if status is not ready yet